File: javascript/module2/solutions/solution5.js
function getTotalArmyPower(numOfSwordsmen, numOfKnights, kingPower) {
    const minKingPowerToInculdeOtherTypes = 5;
    const infantryMultiplier = 10;
    const knightsMultiplier = 20;
    
    if (kingPower <= minKingPowerToInculdeOtherTypes) {
      return kingPower;
    }
  
    const infantryPower = getPower(numOfSwordsmen, infantryMultiplier);
  
    const knightsPower = getPower(numOfKnights, knightsMultiplier);
  
    return kingPower + infantryPower + knightsPower;
  }
  
  function getPower(numOfUnits, multiplier) {
    const minPowerForBonus = 100;
    const powerBonus = 50;
  
    let result = numOfUnits * multiplier;
    if (result >= minPowerForBonus) {
      result += powerBonus;
    }
    return result;
  }

File: csharp/module2/solutions/solution5.cs
public class ArmyPowerCalculator
{
    private const int MinKingPowerToIncludeOtherTypes = 5;
    private const int InfantryMultiplier = 10;
    private const int KnightsMultiplier = 20;

    public int GetTotalArmyPower(int numOfSwordsmen, int numOfKnights, int kingPower)
    {
        if (kingPower <= MinKingPowerToIncludeOtherTypes)
        {
            return kingPower;
        }

        int infantryPower = GetPower(numOfSwordsmen, InfantryMultiplier);
        int knightsPower = GetPower(numOfKnights, KnightsMultiplier);

        return kingPower + infantryPower + knightsPower;
    }

    private int GetPower(int numOfUnits, int multiplier)
    {
        const int MinPowerForBonus = 100;
        const int PowerBonus = 50;

        int result = numOfUnits * multiplier;
        if (result >= MinPowerForBonus)
        {
            result += PowerBonus;
        }
        return result;
    }
}
